lenskit-core/src/main/java/org/lenskit/data/dao/file/CollectionEntitySource.java

```diff
@@ -26,12 +26,14 @@
 import org.lenskit.util.io.ObjectStreams;
 
 import java.io.IOException;
+import java.io.Serializable;
 import java.util.Collection;
 
 /**
  * Entity source backed by a collection.
  */
-class CollectionEntitySource implements EntitySource {
+class CollectionEntitySource implements EntitySource, Serializable {
```

Comment:
The `file` should probably be renamed to `static` - it really encompasses a family static data sources.

For iterative methods, I think you are right that we need something else. Perhaps something like the `PreferenceSnapshot` but more generic. It should live on top of the DAOs, and have a fast path to reuse the in-memory data strategy employed by the static file DAO provider (which a subsequent PR is renaming to `StaticDataSource`).

`EntitySource` is not intended to be a very public component - it's mostly an implementation detail of the static DAO. Algorithms that depend on it will not work with any other DAOs, such as a live database DAO. It is public because the evaluator needs to be able to reach into the internal components of the static DAO provider.